### PR TITLE
Add support for `if TYPE_CHECKING is True` operation

### PR DESCRIPTION
**Pull request by iamibi**
Fixes https://github.com/snok/flake8-type-checking/issues/156

### CODE REVIEW

**Comment by sondrelg** [comment]
I'll try to review this, this weekend 👍

File: flake8_type_checking/checker.py [hunk 1]
```diff
@@ -485,6 +485,41 @@ def is_type_checking(self, node: ast.AST) -> bool:
...
+        # Right side of the operator should be a constant `True`.
+        right_value_true = (
+            hasattr(node, 'comparators') is True
+            and len(node.comparators) == 1
+            and isinstance(node.comparators[0], ast.Constant) is True
+            and hasattr(node.comparators[0], 'value') is True
+            and node.comparators[0].value is True
+        )
+
+        return type_checking_on_left and operator_is and right_value_true
```
Comment (sondrelg) [inline flake8_type_checking/checker.py:521]: Very nice work :+1: 

I think there's a few conditions here we can leave out, like the `hasattr(..., 'value')` for `ast.Constant` and similar checks. These will always hold I believe. Taking that into consideration, and re-using the `is_type_checking` method from above this, I believe we might be able to reduce the code to this:

```suggestion
def is_type_checking_true(self, node: ast.Compare) -> bool:
    """
    An ast.Compare node has a `left`, `ops`, and `comparators` attribute.

    Here we want to check whether our node corresponds to
    
        `if TYPE_CHECKING is True`
                ^         ^    ^____ comparators
       left ____|       ops
    """
    # Left side should be a TYPE_CHECKING block
    if not hasattr(node, 'left') or not self.is_type_checking(node.left):
        return False
    
    # Operator should be `is`
    if not len(node.ops) == 1 or not isinstance(node.ops[0], ast.Is):
        return False
    
    # Right side should be `True`
    if not len(node.comparators) == 1 or not isinstance(node.comparators[0], ast.Constant) or not node.comparators[0].value is True:
        return False

    return True
```

**Comment by sondrelg** [comment]
Thanks @iamibi. New version should be out now :tada:

**Comment by iamibi** [comment]
> Thanks @iamibi. New version should be out now 🎉

Thanks @sondrelg for the help.

**Pull request merged**